### ISSUE TITLE
Issue using with LazyVim

### ISSUE THREAD

**Issue by JakeHandsome** [NicoElbers/nv#14]
I tried setting up this project on my neovim config that uses [LazyVim](https://www.lazyvim.org/) and was running into issue of getting nv to properly load the plugins from the nix store (except for lazy-nivm and LazyVim) 

I know in my example I didn't have every single plugin in the flake, but I did include `lualine-nvim` but lazy is showing it loaded from the outside the nix store path
![image](https://github.com/user-attachments/assets/cb724cfc-c654-4c11-9a62-b0a46cd309ff)

I also get this error that my directory overwriting the lazyvim defaults was unable to be loaded

![image](https://github.com/user-attachments/assets/caa1ddf1-ecbd-4eb0-bb47-ff04ee4aaf04)

I have re-created the issues using the LazyVim starter template 

Steps to reproduce
1. clone https://github.com/JakeHandsome/nv-lazy-starter
1. run `nix build`
2. run `./result/bin/nv`
3. run `:Noice history` in neovim to view the error

**Comment by NicoElbers** [comment]
Hai, thanks for the issue. LazyVim in particular (although I suspect this is true for every distribution) is very complex. Nv should technically work for them, but they use a lot of archane magic that I simply haven't looked at.

However I do want to support LazyVim, I'm planning to first write a tool that can auto gen the flake (so I don't have to go through the entirely of LazyVim), and afterwards I want to get nv working on LazyVim with minimal to no patches.

All that to say, if you're really invested, I'd love to help you debug this, but personally, I'll have a look at it within the coming months (Uni is busy)

**Comment by JakeHandsome** [comment]
> All that to say, if you're really invested, I'd love to help you debug this, but personally, I'll have a look at it within the coming months (Uni is busy)

Totally understand if you are busy. I would love to help figure out this issue if you have any ideas of where to look first

**Comment by NicoElbers** [comment]
In my quick look yesterday I found that your plugin directory isn't being patched. I have a feeling that this is because there's a `'` in a comment, which the parser then assumes is the start of a string and in turn it ignores the rest of the file. 

There are 2 issues hidden in that
1) if a `'`, `"` or `[[` is found in a comment it should be ignored

2) if a `'` or `"` is found in a line and are not closed in that line, they should be ignored. Although I'll have to check with the lua langref

Along side this, an arguably bigger problem is that LazyVim exposes a spec of plugins. In other words, the spec isn't in your config and thus nv can't patch it.

The solution to the second problem is solvable if I maintain a fork, which I would be willing to do if I can patch everything in one go, but Im not there yet.

TLDR; there's a very solvable problem here that you can hack on if you like, but I don't think you'll be able to use any distribution (In a nix way) with nv until the project is further along.

**Comment by NicoElbers** [comment]
Relevant lua language reference:
> A comment starts with a double hyphen (--) anywhere outside a string. If the text immediately after -- is not an opening long bracket, the comment is a short comment, which runs until the end of the line. Otherwise, it is a long comment, which runs until the corresponding closing long bracket. Long comments are frequently used to disable code temporarily.
